lib/cloudregister/registerutils.py

```diff
@@ -481,8 +486,7 @@ def get_instance_data(config):
     if instance_data:
         # Marker for the server to not return https:// formated
         # service and repo information
-        instance_data += b'<repoformat>plugin:susecloud</repoformat>\n'
-        return instance_data.decode()
+        instance_data += '<repoformat>plugin:susecloud</repoformat>\n'
```

Comment:
The client probably should always send this, even if the metadata is not available. Otherwise if the client can't fetch metadata, the server would fall back to HTTPS URLs, even if the the client is recent.

Reply:
OK, that makes sens and simplifies things. That means this function always has a return value and all the other if conditions go away :) Also centralized the logging of the warning message. Great :)